lib/crasher.js

```diff
@@ -33,19 +33,35 @@ function logCrashAndExit (logPath, errorInfo, checkout, stackTrace, err) {
   }
 }
 
-function setupCrashHandlers (processName, swap) {
+function printAndExit (errorInfo, stackTrace, err) {
+  // Electron does not play well with process.exit, so use app.exit instead
+  const runContext = isBare ? global.Bare : (global.process.versions.electron ? require('electron').app : global.process)
+  const pid = isBare ? Bare.pid : global.process.pid
+  const errInfo = err !== null && typeof err === 'object' ? JSON.stringify(err, 0, 4).slice(1, -2) : ''
+  const errorMsg = `${stackTrace + errInfo}\n\n${errorInfo}`
+
+  console.error(errorMsg)
+
+  if (isBare) {
+    os.kill(pid)
+  } else {
+    runContext.exit(1)
+  }
+}
+
+function setupCrashHandlers (processName, swap, enableLog) {
   const crashlogPath = path.join(swap, `${processName}.crash.log`)
   const runContext = isBare ? global.Bare : global.process
 
   runContext.on('unhandledRejection', (reason) => {
     const stack = reason?.stack || reason || ''
     const errorInfo = `${processName} exiting due to unhandled rejection`
-    logCrashAndExit(crashlogPath, errorInfo, checkout, stack, reason)
+    enableLog ? logCrashAndExit(crashlogPath, errorInfo, checkout, stack, reason) : printAndExit(errorInfo, stack, reason)
```

Comment:
I think we should follow the code style convention and use if-else instead of ?

Reply:
updated